### PR TITLE
Torchgeo multitask

### PR DESCRIPTION
**Pull request by tlarcher**
# Reference Issues/PRs

# What does this implement/fix? Explain your changes.
## Major improvements

- Added easy and straightforward classification task selection for deep learning through configuration files for `RasterTorchGeoDataset`datasets
- Added the option to download sample data for example `cnn_on_rgbnir_torchgeo.py`
- Fixed inference crash when loading a checkpoint via pytorchlightning 
- Fixed type selection for `CrossEntropyLoss` and `BCEWithLogitsLoss` depending on the deep learning task selected

## Minor improvements
- Added and updated docstrings
- Cleaned unused imports.

# Any other comments?

### CODE REVIEW

**Comment by tlarcher** [comment]
Torchgeo multitask

- Added easier task switch through config file (updated only torchgeo example file).
- Modified torchgeo_dataset to correctly type labels in accordance to the task (especially the accuracy metrics).
- Modified standard_prediction_system to correctly type the y targets for the losses (which can be different from the types required 
- Added download option to download Sentinel-2A sample data (4 bands of 1 tile over Montpellier) from Microsoft Planetary Computer since these tiles are weigh a lot and cannot be pushed on the remote repository.
- Fixed inference crashing when using new hparams preprocessing in main script ClassificationSystem constructor.
- Added hparams preprocessing in main script ClassificationSystem constructor.
- Renamed RasterTorchGeoDataset constructor arguments and removed unused variables.
- Added boolean argument in ClassificationSystem constructor so as to not triger the hparams preprocessing when loading weights from a checkpoint because pytorchlightning runs the constructor of the class on which  is called.
by the metrics.
- Added doctrings.
- Cleaned unused imports.
